File: documentation-examples-kotlin/src/kotlin/dev/typr/foundations/docs/postgresql/NullableTypes.kt
package dev.typr.foundations.docs.postgresql

import dev.typr.kotlinfoundations.*
import dev.typr.kotlinfoundations.data.*

@Suppress("unused")
class NullableTypes {
    //start
    val notNull: PgType<Int> = PgTypes.int4
    val nullable: PgType<Int?> = PgTypes.int4.opt()
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/NullableTypes.scala
package dev.typr.foundations.docs.postgresql
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*



@SuppressWarnings(Array("unused"))
object NullableTypes:
  //start
  val notNull: PgType[Int] = PgTypes.int4
  val nullable: PgType[Option[Int]] = PgTypes.int4.opt
  //stop
